engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/exception/KafkaJsonExceptionSerializationSchema.scala

```diff
@@ -0,0 +1,69 @@
+package pl.touk.nussknacker.engine.kafka.exception
+
+import io.circe.generic.JsonCodec
+import io.circe.syntax.EncoderOps
+import org.apache.kafka.clients.producer.ProducerRecord
+import pl.touk.nussknacker.engine.api.MetaData
+import pl.touk.nussknacker.engine.api.exception.{EspExceptionInfo, NonTransientException}
+
+import java.io.{PrintWriter, StringWriter}
+import java.net.InetAddress
+import java.nio.charset.StandardCharsets
+import java.time.ZoneId
+import scala.io.Source
+
+case class KafkaJsonExceptionSerializationSchema(metaData: MetaData, consumerConfig: KafkaExceptionConsumerConfig) {
+
+  def serialize(exceptionInfo: EspExceptionInfo[NonTransientException]): ProducerRecord[Array[Byte], Array[Byte]] = {
+    val key = s"${metaData.id}-${exceptionInfo.nodeId.getOrElse("")}".getBytes(StandardCharsets.UTF_8)
+    val value = KafkaExceptionInfo(metaData, exceptionInfo, consumerConfig)
+    val serializedValue = value.asJson.spaces2.getBytes(StandardCharsets.UTF_8)
+    new ProducerRecord(consumerConfig.topic, key, serializedValue)
+  }
+
+}
+
+@JsonCodec case class KafkaExceptionInfo(processName: String,
+                                         nodeId: Option[String],
+                                         message: Option[String],
+                                         exceptionInput: Option[String],
+                                         //TODO: consider using JSON here?
+                                         inputEvent: Option[String],
+                                         stackTrace: Option[String],
+                                         timestamp: Long,
+                                         host: Option[String],
+                                         additionalData: Map[String, String])
+
+object KafkaExceptionInfo {
+
+  //TODO: better hostname (e.g. from some Flink config)
+  private lazy val hostName = InetAddress.getLocalHost.getHostName
+
+  def apply(metaData: MetaData, exceptionInfo: EspExceptionInfo[NonTransientException], config: KafkaExceptionConsumerConfig): KafkaExceptionInfo = {
+    new KafkaExceptionInfo(
+      metaData.id,
+      exceptionInfo.nodeId,
+      Option(exceptionInfo.throwable.message),
+      Option(exceptionInfo.throwable.input),
+      optional(exceptionInfo.context, config.includeInputEvent).map(_.toString),
+      serializeStackTrace(config.stackTraceLengthLimit, exceptionInfo.throwable),
```

Comment:
I see one problem here. We have two possible cases of `NonTransientException` - one is class which extends  `NonTransientException` and the other one case (used more often), when we wrap some `Exception` with `NonTransientException`. In the second case IMO will be seen in stacktrace only `Unknown exception`, because default stack trace limit is set to low limit (10). I wonder if we should have some extraction logic that will extract "interesting" exception. This handler could skip `NonTransientException` that is only the wrapper and some other common "wrapper" exceptions like `java.util.concurrent.{CompletionException, ExecutionException}`

Reply:
We already skip some common stuff in `DeeplyCheckingExceptionExtractor` - but only to get to NonTransientException. I think that detecting "real" cause can be not so simple. 
I think we can raise default limit (like, to 50?) and do some work to throw NonTransientException when we expect they can happen (e.g. with spel evaluation) instead of catching them in DefaultExceptionHandler
WDYT?

Reply:
ahh, indeed! yes, please raise default limit

Reply:
done